gstreamer-pbutils/src/auto/discoverer.rs

```diff
@@ -0,0 +1,149 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir @ 47eb915)
+// from gir-files (https://github.com/gtk-rs/gir-files @ ???)
+// DO NOT EDIT
+
+use DiscovererInfo;
+use Error;
+use ffi;
+use glib;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use gst;
+use gst_ffi;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct Discoverer(Object<ffi::GstDiscoverer, ffi::GstDiscovererClass>);
+
+    match fn {
+        get_type => || ffi::gst_discoverer_get_type(),
+    }
+}
+
+impl Discoverer {
+    pub fn new(timeout: gst::ClockTime) -> Result<Discoverer, Error> {
+        assert_initialized_main_thread!();
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret = ffi::gst_discoverer_new(timeout.to_glib(), &mut error);
+            if error.is_null() { Ok(from_glib_full(ret)) } else { Err(from_glib_full(error)) }
+        }
+    }
+}
+
+unsafe impl Send for Discoverer {}
+unsafe impl Sync for Discoverer {}
+
+pub trait DiscovererExt {
+    fn discover_uri(&self, uri: &str) -> Result<DiscovererInfo, Error>;
+
+    fn discover_uri_async(&self, uri: &str) -> Result<(), glib::error::BoolError>;
+
+    fn start(&self);
+
+    fn stop(&self);
+
+    fn connect_discovered<F: Fn(&Self, &DiscovererInfo, &Option<Error>) + Send + Sync + 'static>(&self, f: F) -> SignalHandlerId;
+
+    fn connect_finished<F: Fn(&Self) + Send + Sync + 'static>(&self, f: F) -> SignalHandlerId;
+
+    fn connect_source_setup<F: Fn(&Self, &gst::Element) + Send + Sync + 'static>(&self, f: F) -> SignalHandlerId;
+
+    fn connect_starting<F: Fn(&Self) + Send + Sync + 'static>(&self, f: F) -> SignalHandlerId;
+}
+
+impl<O: IsA<Discoverer> + IsA<glib::object::Object>> DiscovererExt for O {
+    fn discover_uri(&self, uri: &str) -> Result<DiscovererInfo, Error> {
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret = ffi::gst_discoverer_discover_uri(self.to_glib_none().0, uri.to_glib_none().0, &mut error);
+            if error.is_null() { Ok(from_glib_full(ret)) } else { Err(from_glib_full(error)) }
+        }
+    }
+
+    fn discover_uri_async(&self, uri: &str) -> Result<(), glib::error::BoolError> {
+        unsafe {
+            glib::error::BoolError::from_glib(ffi::gst_discoverer_discover_uri_async(self.to_glib_none().0, uri.to_glib_none().0), "Failed to add URI to list of discovers")
+        }
+    }
+
+    fn start(&self) {
+        unsafe {
+            ffi::gst_discoverer_start(self.to_glib_none().0);
+        }
+    }
+
+    fn stop(&self) {
+        unsafe {
+            ffi::gst_discoverer_stop(self.to_glib_none().0);
+        }
+    }
+
+    fn connect_discovered<F: Fn(&Self, &DiscovererInfo, &Option<Error>) + Send + Sync + 'static>(&self, f: F) -> SignalHandlerId {
```

Comment:
DiscovererInfo is always created, so it is returned even if with no information at all.